lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx

```diff
@@ -4,167 +4,92 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React from 'react';
-import { shallow } from 'enzyme';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faCheck } from '@fortawesome/free-solid-svg-icons';
+import React, { RefAttributes } from 'react';
+import { within } from '@testing-library/react';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+import userEvent from '@testing-library/user-event';
 
 import NxCheckbox, { Props } from '../NxCheckbox';
-import NxOverflowTooltip from '../../NxTooltip/NxOverflowTooltip';
 
 describe('NxCheckbox', function() {
-  const simpleProps: Props = {
-    checkboxId: 'checkbox-id',
-    isChecked: false,
-    onChange: () => {},
-    disabled: undefined,
-    children: undefined
-  };
-
-  function getShallowComponent(additionalProps?: Partial<Props>) {
-    return shallow(<NxCheckbox { ...simpleProps } { ...additionalProps } />);
-  }
-
-  it('renders a <label> containing a checkbox <input> and visible box', function() {
-    const shallowRender = getShallowComponent();
-
-    expect(shallowRender).toMatchSelector('label.nx-checkbox');
-
-    expect(shallowRender.find('input')).toHaveProp('type', 'checkbox');
-    expect(shallowRender.find('input')).toHaveProp('id', 'checkbox-id');
-    expect(shallowRender.find('input')).toHaveClassName('nx-checkbox__input');
-
-    expect(shallowRender.find('.nx-checkbox__box')).toExist();
-
-    // space is nbsp for empty box
-    expect(shallowRender).toHaveText(' ');
+  const minimalProps: Props & RefAttributes<HTMLLabelElement> = {
+        isChecked: false
+      },
+      quickRender = rtlRender(NxCheckbox, minimalProps),
+      renderEl = rtlRenderElement(NxCheckbox, minimalProps);
+
+  it('renders a <label> containing a checkbox role <input>', function() {
+    const component = renderEl()!,
+        input = within(component).getByRole('checkbox');
+
+    expect(component.tagName).toBe('LABEL');
```

Comment:
I tend to think the label element is an implementation detail.  We do want to check that it gets connected as the accessible name of the input though.

Reply:
Updated here
a42ddb52a04ffc150ea2260c27f66d569f93b25c
b6036817b68162b098ddabae1831cb0171dd3cbe